ui/src/app/components/cromwell-configuration-panel.tsx

```diff
@@ -0,0 +1,245 @@
+import * as React from 'react';
+import { useEffect } from 'react';
+import * as fp from 'lodash/fp';
+
+import { AppStatus, AppType, UserAppEnvironment } from 'generated/fetch';
+
+import { Button } from 'app/components/buttons';
+import { FlexColumn, FlexRow } from 'app/components/flex';
+import { WarningMessage } from 'app/components/messages';
+import { styles } from 'app/components/runtime-configuration-panel/styles';
+import { Spinner } from 'app/components/spinners';
+import { appsApi } from 'app/services/swagger-fetch-clients';
+import {
+  withCdrVersions,
+  withCurrentWorkspace,
+  withUserProfile,
+} from 'app/utils';
+import {
+  DEFAULT_MACHINE_NAME,
+  findMachineByName,
+  Machine,
+} from 'app/utils/machines';
+import { setSidebarActiveIconStore } from 'app/utils/navigation';
+
+import { defaultCromwellConfig, findApp, UIAppType } from './apps-panel/utils';
+import { EnvironmentInformedActionPanel } from './environment-informed-action-panel';
+
+const { useState } = React;
+
+const cromwellSupportArticles = [
+  { text: 'How to run Cromwell in All of Us workbench?', link: '#' },
+  { text: 'Cromwell documentation', link: '#' },
+  { text: 'Workflow and WDL', link: '#' },
+  { text: 'Running and Autopause', link: '#' },
+  { text: 'Storage options', link: '#' },
+];
+const DEFAULT_MACHINE_TYPE: Machine = findMachineByName(DEFAULT_MACHINE_NAME);
+
+const { cpu, memory } = DEFAULT_MACHINE_TYPE;
+
+const PanelMain = fp.flow(
+  withCdrVersions(),
+  withCurrentWorkspace(),
+  withUserProfile()
+)(
+  ({
+    analysisConfig,
+    workspace,
+    profileState,
+    creatorFreeCreditsRemaining,
+    onClose,
+  }) => {
+    // all apps besides Jupyter
+    const [userApps, setUserApps] = useState<UserAppEnvironment[]>();
+    const [creationStarted, setCreationStarted] = useState(false);
+    const [loading, setLoading] = useState(true);
+    const [status, setStatus] = useState<AppStatus>();
+
+    useEffect(() => {
+      appsApi().listAppsInWorkspace(workspace.namespace).then(setUserApps);
+    }, []);
+
+    useEffect(() => {
+      if (userApps !== undefined) {
+        const app = findApp(userApps, UIAppType.CROMWELL);
+        if (app) {
+          setStatus(app.status ?? null);
+        } else {
+          setLoading(false);
+        }
+      }
+    }, [userApps]);
+
+    useEffect(() => {
+      if (loading && status !== undefined) {
+        setLoading(false);
+      }
+    }, [loading, status]);
+
+    useEffect(() => {
+      if (creationStarted) {
+        appsApi().createApp(workspace.namespace, defaultCromwellConfig);
+        onClose();
+        setTimeout(() => setSidebarActiveIconStore.next('apps'), 3000);
+      }
+    }, [creationStarted]);
```

Comment:
There is a lot of state and `useEffect` calls here. Could it be simplified? Maybe something like the following:

```
const [creationStarted, setCreationStarted] = useState(false);
const status = findApp(userApps, UIAppType.CROMWELL);
const loading = !app.status;

useEffect(() => {
  appsApi().listAppsInWorkspace(workspace.namespace).then(setUserApps);
}, []);
```

This should update the component every time `findApp` changes value.

Reply:
Additionally, it might be simpler to move the fourth `useEffect` into the function that calls `setCreationStarted(true)`.

Reply:
Good call. I like this refactor. Thanks for the suggestion.